bin/main.go

```diff
@@ -19,13 +20,32 @@ var connect = flag.String("connect", "",
 var autodiscover = flag.Bool("auto", false,
 	"Whether we should try and find neighboring routers")
 
-func FindNeighbors(dev net.Interface, key node.PublicKey) <-chan node.NodeAddress {
+func GetLinkLocalAddr(dev net.Interface) (*net.IPAddr, error) {
+	cidr_ll_addr, err := dev.Addrs()
+	if err != nil {
+		return nil, err
+	}
+
+	ll_addr, _, err := net.ParseCIDR(cidr_ll_addr[1].String())
```

Comment:
I tested this in my development VM and my VPS and the second one always appears to be a link local address. Do you want to test it on a couple machines before I merge?